src/drivers/QtTestDriver.cpp

```diff
@@ -0,0 +1,203 @@
+#include "cucumber-cpp/internal/drivers/QtTestDriver.hpp"
+
+#include <QtTest/QtTest>
+
+/* Based on Sir Digby Chicken post from:
+ * http://stackoverflow.com/questions/5419356/redirect-stdout-stderr-to-a-string
+ * who granted his work on BSD license
+ * adjusted for Qt by konserw
+ */
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32) && !defined(__CYGWIN__)
+#define WINDOWS
+#include <io.h>
+#include <fcntl.h>
+#define popen _popen 
+#define pclose _pclose
+#define stat _stat 
+#define dup _dup
+#define dup2 _dup2
+#define fileno _fileno
+#define close _close
+#define pipe _pipe
+#define read _read
+#define eof _eof
+#else
+#include <errno.h>
+#endif
+
+#include <cstdio>
+#include <QtConcurrent/QtConcurrent>
+
+#ifndef STD_OUT_FD
+#define STD_OUT_FD (fileno(stdout))
+#endif
+#ifndef STD_ERR_FD
+#define STD_ERR_FD (fileno(stderr))
+#endif
+
+class qtCapture {
+public:
+    qtCapture(): m_capturing(false) {
+        // make stdout & stderr streams unbuffered
+        // so that we don't need to flush the streams
+        // before capture and after capture 
+        // (fflush can cause a deadlock if the stream is currently being 
+        QMutexLocker locker(&m_mutex);
+        setvbuf(stdout, NULL, _IONBF, 0);
+        setvbuf(stderr, NULL, _IONBF, 0);
+    }
+    ~qtCapture() {
+        if(m_capturing)
+            EndCapture();
+    }
+
+    void BeginCapture() {
+        QMutexLocker locker(&m_mutex);
+        if (m_capturing)
+            return;
+
+        secure_pipe(m_pipe);
+        m_oldStdOut = secure_dup(STD_OUT_FD);
+        m_oldStdErr = secure_dup(STD_ERR_FD);
+        secure_dup2(m_pipe[WRITE], STD_OUT_FD);
+        secure_dup2(m_pipe[WRITE], STD_ERR_FD);
```

Comment:
This has one rather fatal flaw: it won't work if whatever you write more than the pipe's buffer can hold to either of these file descriptors. At that point it _will_ block and you're process will hang.